cloudflare/src/framework/blocking_api.rs

```diff
@@ -0,0 +1,72 @@
+use reqwest::blocking::RequestBuilder;
+use serde::Serialize;
+
+use crate::framework::auth::Credentials;
+use crate::framework::reqwest_adaptors::match_reqwest_method;
+use crate::framework::{
+    apiclient::ApiClient, auth, auth::AuthClient, endpoint, response, response::map_api_response,
+    Environment, HttpApiClient, HttpApiClientConfig,
+};
+
+impl HttpApiClient {
+    pub fn new(
+        credentials: auth::Credentials,
+        config: HttpApiClientConfig,
+        environment: Environment,
+    ) -> anyhow::Result<HttpApiClient> {
+        let http_client = reqwest::blocking::Client::builder()
+            .timeout(config.http_timeout)
+            .default_headers(config.default_headers)
+            .build()?;
+
+        Ok(HttpApiClient {
+            environment,
+            credentials,
+            http_client,
+        })
+    }
+}
+
+// TODO: This should probably just implement request for the Reqwest client itself :)
+// TODO: It should also probably be called `ReqwestApiClient` rather than `HttpApiClient`.
```

Comment:
I wouldn't know, maybe @adamchalmers  can chime in.

Based on his feedback about the other stuff, I think we'd prefer to not break symbols unnecessarily that would cause users to have to adapt their code when updating cloudflare-rs version.
Still, my guess is only as good as yours (or even less).

But given that I'm only moving the implementation to a new file, as the whole PR is a non-breaking mini-refactor from the perspective of non-wasm targets, I did not want to meddle with what already existed here.